File: server/test-gemini.js
import { GoogleGenAI } from '@google/genai';
import dotenv from 'dotenv';

dotenv.config();

const API_KEY = process.env.GEMINI_API_KEY;

console.log('Testing Gemini API with Node.js...');
console.log('API Key:', API_KEY ? `${API_KEY.substring(0, 10)}...` : 'NOT FOUND');

const ai = new GoogleGenAI({ apiKey: API_KEY });

async function testGemini() {
  try {
    console.log('\nSending test prompt with gemini-3-flash-preview...');
    
    const response = await ai.models.generateContent({
      model: "gemini-3-flash-preview",
      contents: 'Say "Hello, Kai Placement Copilot is working!" in one sentence.'
    });
    
    console.log('✅ Success!');
    console.log('Response:', response.text);
  } catch (error) {
    console.error('❌ Error:', error.message);
    if (error.response) {
      console.error('Response:', error.response);
    }
  }
}

testGemini();

File: test_gemini.py
#!/usr/bin/env python3
import os
from google import genai

# Set API key
os.environ['GEMINI_API_KEY'] = os.environ.get('GEMINI_API_KEY', 'your_gemini_api_key_here')

print("Testing Gemini API with correct model...")

client = genai.Client(api_key=os.environ['GEMINI_API_KEY'])

# Test with gemini-3-flash-preview (the model user mentioned)
print("\n1. Testing gemini-3-flash-preview...")
try:
    response = client.models.generate_content(
        model='gemini-3-flash-preview',
        contents='Say "Hello, Kai Placement Copilot is working!" in one sentence.'
    )
    print(f"✅ Success with gemini-3-flash-preview!")
    print(f"Response: {response.text}")
except Exception as e:
    print(f"❌ Error: {e}")

# Test with gemini-2.0-flash
print("\n2. Testing gemini-2.0-flash...")
try:
    response = client.models.generate_content(
        model='gemini-2.0-flash',
        contents='Say "Hello, Kai Placement Copilot is working!" in one sentence.'
    )
    print(f"✅ Success with gemini-2.0-flash!")
    print(f"Response: {response.text}")
except Exception as e:
    print(f"❌ Error: {e}")

# Test with gemini-flash-latest
print("\n3. Testing gemini-flash-latest...")
try:
    response = client.models.generate_content(
        model='gemini-flash-latest',
        contents='Say "Hello, Kai Placement Copilot is working!" in one sentence.'
    )
    print(f"✅ Success with gemini-flash-latest!")
    print(f"Response: {response.text}")
except Exception as e:
    print(f"❌ Error: {e}")
